### ISSUE TITLE
Deep Short Text Classification with Knowledge Powered Attention(2019)

### ISSUE THREAD

**Issue by magicpieh28** [magicpieh28/Paper-Summary#18]
# about this paper
Author: Jindong Chen, Yizhou Hu, Jingping Liu, Yanghua Xiao, Haiyun Jiang
Link: https://arxiv.org/pdf/1902.08050.pdf
##
短文の分類は情報不足が原因でうまくできないけどKBsを用いたattentionモデルで解決するという論文。

# aim
## problems
- explicit model: explicit表現は一般的に短文の文脈を無視し、深い意味情報をつかみ出すことができない
- implicit model: implicit表現はKBsの中のisA, isPropertyOFのような大事な意味的関係を無視する
- KBs
  - "Alice has been using Apple for more than 10 years": この文でのAppleは果物ではないブランドを指していることは文を読めばわかるけど、KBsの中で適切なコンセプトを持ってこれない問題
  - "Bill Gates is one of the co-founders of Microsoft": 人と企業家というコンセプトが同時に存在している場合、より詳しい情報のために企業家というコンセプトを持ってきてほしいけどそうできない問題

## how to solve
explicitかつimplicitなモデルを使用する。

# model
![image](https://user-images.githubusercontent.com/40426884/53697141-95cb3700-3e11-11e9-807a-37a0ade8f1a3.png)

モデルを構成するモジュールは4つ。
- KBsから情報（知識）を持ってくる
- 入力の短文をエンコード: 文の表現q
- KBsから抽出したコンセプトをエンコード: コンセプト表現p
- qとpをコンキャットし分布を算出

# experiments
## datasets
- train:
  - NLPCC2013: Chinese Weibo emotion analysis
  - NLPCC2014: product review
  - NLPCC2017: Chinese news title with 18 classes (e.g. entertainment, game, food)

短文の長さの平均は12。

- test:
  - Sogou news (Fu et al. 2015) (e.g. military, politics)

## settings
optimizer: Adam
learning rate: 0.01
batch size: 64
epoch: 20
embedding dimension: 50 (pre-trained on Sogou News corpus)
filter: [2, 3, 4] (size: 50 for each)
- test set:
  - u = 64
  - d_a = 70
  - d_b = 35
  - γ = ニューラルネットワークから自動学習

- evaluation metric: accuracy

## results
![image](https://user-images.githubusercontent.com/40426884/53697274-4a198d00-3e13-11e9-9b33-45bd60448f27.png)

![image](https://user-images.githubusercontent.com/40426884/53697278-59003f80-3e13-11e9-86f9-07eb48b55b8e.png)

γは0.25の時が最も良かった。

![image](https://user-images.githubusercontent.com/40426884/53697298-8d73fb80-3e13-11e9-82b7-c91a9b183070.png)

STCKA-non-static: あらかじめ訓練されたembeddingを初期値として使い、訓練段階でチューニング

レアーな単語と未知語に対していい性能を見せた。

## error
long-tailed（おそらく以前からのさらなる文脈情報がないといけないという意味）entityに対してはいい性能が出なかった。
短文の中で言及されていないentityに関してはコンセプトを持ってくることさえできなかったのでいい性能は当たり前に見せられなかった。例えば、"don't pay money, it's all routines"のような文はファッションカテゴリーに分類されるべきだが、そうできなかった。

# next
[A structured self-attentive sentence embedding. 2017](https://arxiv.org/pdf/1703.03130.pdf)
[Attention is all you need. 2017](https://papers.nips.cc/paper/7181-attention-is-all-you-need.pdf)
